dae/package.nix

```diff
@@ -1,47 +1,39 @@
-{ clang
-, fetchFromGitHub
-, buildGoModule
-, lib
-}:
-buildGoModule (
-  let dyn = lib.strings.splitString "\n" (builtins.readFile ./dynamic_info); in rec {
-    pname = "dae";
-    version = with lib; substring 0 8 (elemAt dyn 0);
-
-    src = fetchFromGitHub {
-      owner = "daeuniverse";
-      repo = pname;
-      rev = lib.elemAt dyn 0;
-      hash = lib.elemAt dyn 1;
-      fetchSubmodules = true;
-    };
-
-    vendorHash = lib.elemAt dyn 2;
-
-    proxyVendor = true;
-
-    nativeBuildInputs = [ clang ];
-
-    ldflags = [
-      "-s"
-      "-w"
-      "-X github.com/daeuniverse/dae/cmd.Version=${version}"
-      "-X github.com/daeuniverse/dae/common/consts.MaxMatchSetLen_=64"
-    ];
-
-    preBuild = ''
-      make CFLAGS="-D__REMOVE_BPF_PRINTK -fno-stack-protector" \
-      NOSTRIP=y \
-      ebpf
-    '';
-
-    # network required
-    doCheck = false;
-
-    postInstall = ''
-      install -Dm444 install/dae.service $out/lib/systemd/system/dae.service
-      substituteInPlace $out/lib/systemd/system/dae.service \
-        --replace /usr/bin/dae $out/bin/dae
-    '';
-  }
-)
+src: { clang
+     , fetchFromGitHub
+     , buildGoModule
+     , lib
+     }:
+buildGoModule rec {
+  pname = "dae";
+  version = src.shortRev;
+
+  inherit src;
+
+  vendorHash = "sha256-iVhIpnnXGDop9meShktheb0tmI+ytFV5PF0PK9CHeTY=";
```

Comment:
Do we need to maintain this sha in the future? Can it be automated?

Reply:
> Do we need to maintain this sha in the future? Can it be automated?

This requires manual changes when dependencies change, not sure if there is an automated way to do this at the moment.

Reply:
> Instead of fetch from input with flake = false, you can use normal fetchFromGitHub and add a passthru.updateScript = nix-update-script { };
> 
> nix-update will then take care of everything including the vendorsha
> 
> https://github.com/NixOS/nixpkgs/blob/09f2d76574f99b2334c07b947ab61ead666c48d4/pkgs/tools/misc/dsq/default.nix

Thanks! I will update a commit with this suggestion.